### ISSUE TITLE
Issue unsubscribing from event

### ISSUE THREAD

**Issue by jonakirke94** [probil/vue-socket.io-extended#400]
I'm dynamically subscribing to an event in my "Created" hook and attempting to unsubscribe in the "BeforeDestroyed" hook. Using the perscribed way in the docs I'm not able to remove the event resulting in the number of subscribtions to the event doubles every time i reload the component..

```js
        created() {
		this.$options.sockets.FAKE = (data) => {
			console.log('RECIEVED FAKE EVENT')
		}
	}

	beforeDestroy() {
		const { sockets = {} } = this.$options;

		console.log(sockets, 'before');
		delete sockets.FAKE;
		console.log(sockets, 'after');
	}
```

I tried doing "sockets.FAKE = null" which resolves the problem of multiple subscriptions but this doesn't seem to be the intended way?

My project is using typescript but as far as I know that shouldn't matter.

**Comment by probil** [comment]
Hi @jonakirke94 👋 
I will take a look. Thanks for your issue!

**Comment by probil** [comment]
Wow. Good catch @jonakirke94 It's a bug!

I will fix and push a new version of the library soon

**Comment by probil** [comment]
@jonakirke94 
Fixed in `v3.2.2`. Please take a look!
